### ISSUE TITLE
java.class.nonPublicPartOfAPI is triggered even if no change has occurred

### ISSUE THREAD

**Issue by mkavanagh** [revapi/revapi#108]
This check doesn't consider whether the same issue afflicted the type in the previous version, so it seems to generate spurious `mvn revapi:check` failures and/or version bumps via `mvn revapi:update-versions` when no change has been made.

I've tried adding this to `passThroughDifferences` for the semver-ignore extension but it won't cause a failure on the first build (due to analysis being skipped), so there doesn't seem to be a way to prevent this from sneaking in to projects (and then causing a spurious breaking change on every build) - and it can sneak in easily through third party dependencies.

**Comment by metlos** [comment]
Hmm, you're right in that it is a little bit sneaky and that it doesn't play well with `update-versions`. The only reason for that is that in my opinion this is quite a serious design issue that should be dealt with.

In case of third party libraries, the check cannot decide whether the library is or is not important for you so it by default just reports the difference the same as for the main checked archives. You can circumvent that by filtering out the archives that you don't want API checked:

```xml
<analysisConfiguration>
  <revapi.filter>
    <archives>
      <exclude>
        <item>com.acme:3rd-party-lib</item>
        <item>another:3rd-party-lib</item>
      </exclude>
    </archives>
  </revapi.filter>
</analysisConfiguration>
```
See https://revapi.org/modules/revapi-basic-features/extensions/filter.html.

**Comment by metlos** [comment]
I think I want to keep the current behavior (unless you convince me with some additional arguments otherwise ;) ) but I will add a configuration property to the check so that it doesn't report if the API hasn't changed since the last version (with the default behavior the same as it is now).

Would that be an acceptable solution for your use case?

**Comment by mkavanagh** [comment]
yep, if it's configurable that solves my issue 👍 

i wonder if it might be worth considering for the future having some checks (I guess this would be the only one right now) that run even if there is no prior version? because that's the other half of the problem, that this defect can sneak into a first version unnoticed

**Comment by metlos** [comment]
Hmm... you're right. The maven plugin doesn't run the analysis if the old or the new version is missing (https://github.com/revapi/revapi/blob/master/revapi-maven-plugin/src/main/java/org/revapi/maven/Analyzer.java#L414).

Could you please create another issue for that so that we can track this problem separately?

**Comment by mkavanagh** [comment]
raised at https://github.com/revapi/revapi/issues/109

**Comment by metlos** [comment]
Implemented in `revapi-java-0.15.0`.
